README.md

```diff
@@ -1,2 +1,75 @@
 # docker-ci-deploy
 Python script to help push Docker images to a registry using CI services
+
+## Installation
+```
+pip install docker-ci-deploy
+```
+
+The script is self-contained and has no dependencies. It can be run by simply executing the [main file](docker-ci-deploy/__main__.py).
+
+## Usage
+The script can tag an existing image, login to a registry, and push the tags to the registry, depending on what arguments are passed to it.
+
+There is one required argument: the image to push.
+
+#### Pushing an image
+```
+docker-ci-deploy my-image:latest
+```
+
+This will simply push the image `my-image:latest` to the default registry (https://hub.docker.com).
+
+#### Logging in
+On a CI service you are unlikely to be logged in to the registry. You can login using the `--login` parameter, which takes an argument of the form `<username>:<password>`.
+```
+docker-ci-deploy --login janedoe:pa$$word my-image:latest
+```
```

Comment:
You need to quote or escape the `$$` otherwise you get the current process id in the middle of your password. ;-)